### ISSUE TITLE
Unexpected parser error in language-docker-6.0.1

### ISSUE THREAD

**Issue by peti** [hadolint/language-docker#37]
I tried to parse https://github.com/peti/opensuse-docker/blob/master/tw-haskell-dev/Dockerfile and got the following error:

~~~
../opensuse-docker/tw-haskell-dev/Dockerfile:18:2:
unexpected '&'
expecting '#', ADD, ARG, CMD, COPY, ENTRYPOINT, ENV, EXPOSE, FROM, HEALTHCHECK, LABEL, MAINTAINER, ONBUILD, RUN, SHELL, STOPSIGNAL, USER, VOLUME, WORKDIR, end of input, or the rest of a new line followed by the next instruction
~~~

Curiously enough, docker parses and handles that Dockerfile just fine, so I reckon this is more likely to be an issue in the parser than an actual error in the file?

**Comment by peti** [comment]
I have `Docker version 18.03.1-ce, build 9ee9f402cd1eba817c5591a64f1d770c87c421a4`.

**Comment by zemanlx** [comment]
hmm I have tried it on Ubuntu VM

```
# wget https://raw.githubusercontent.com/peti/opensuse-docker/master/tw-haskell-dev/Dockerfile -q

# docker --version
Docker version 18.03.1-ce, build 9ee9f40

# docker build -t tw-haskell-dev .
Sending build context to Docker daemon   2.56kB
Error response from daemon: Dockerfile parse error line 18: unknown instruction: &&
```

and it still do not work as I would expect. Looks like dark magic (something not in my knowledge) is happening in your environment.

Anyway oneline to multiline should be broken by using `\` and if that is ignored it sounds like issue with docker/moby or OS and will not work on most of installations.

Can you try it somewhere else? Are you able to reproduce this behaviour?

**Comment by lorenzo** [comment]
well, I guess this was a win for the docker file linter!!

I'm closing as there is no action we need to take here.
